### DISCUSSION TITLE
Make all fonts same X-height for consistency

### DISCUSSION THREAD

**Discussion by ghost** [koreader/koreader#12540] [Ideas]
It is unsupported, but the functionality of this css property: `font-size-adjust` could be nice to make font sizes consistent. It's just an idea I wanted to share, so maybe one day it could be considered.

**Comment by poire-z** [comment]
I didn't know about this CSS property - and it seems quite complex to parse and ensure.
https://developer.mozilla.org/en-US/docs/Web/CSS/font-size-adjust
But for the main advertised purpose - ensuring consistent look of **fallback fonts**, we have an option, enabled by default, that ensures that:
![image](https://github.com/user-attachments/assets/e3150944-3275-44d7-a47d-b301a79079e1)
(and some fixed scaling/adjustment for monospace fonts, which is quite nice when reading inline bits of code).

As for (which is what I understood when first reading your post) having it adjust the size of the default font, so they all got the same x-height when you switch default fonts (which sometimes bother me, sont font have a small x-height, others not, and you need to tune our font size to get the same visually looking size), it's a bit impossible, as you need a reference font - which can only be the default font :) (or the previous default font, but this is a chicken and egg salad).
